internal/commands/push/command_test.go

```diff
@@ -1429,6 +1431,89 @@ func TestPushCommandDeployDraftAndWait(t *testing.T) {
 	})
 }
 
+func TestPushCommandImportDependenciesAndWait(t *testing.T) {
+	groupID, appID, uploadPath := "groupID", "appID", "node_modules.zip"
+	testError := errors.New("something bad happened")
+
+	t.Run("should return an error with a client that fails to import dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+
+		var capturedGroupID, capturedAppID, capturedUploadPath string
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			capturedGroupID = groupID
+			capturedAppID = appID
+			capturedUploadPath = uploadPath
+			return testError
+		}
+		err := importDependenciesAndWait(nil, realmClient, appRemote{groupID, appID}, uploadPath)
+		assert.Equal(t, testError, err)
+
+		t.Log("and should properly pass through the expected inputs")
+		assert.Equal(t, groupID, capturedGroupID)
+		assert.Equal(t, appID, capturedAppID)
+		assert.Equal(t, uploadPath, capturedUploadPath)
+	})
+
+	t.Run("with a client that successfully imports dependencies", func(t *testing.T) {
+		realmClient := mock.RealmClient{}
+		realmClient.ImportDependenciesFn = func(groupID, appID, uploadPath string) error {
+			return nil
+		}
+
+		t.Run("but fails to finish the installation", func(t *testing.T) {
+			realmClient.DependenciesInstallationFn = func(groupID, appID string) (realm.DependenciesInstallation, error) {
+				return realm.DependenciesInstallation{}, testError
+			}
+
+			out, ui := mock.NewUI()
+			err := importDependenciesAndWait(ui, realmClient, appRemote{groupID, appID}, uploadPath)
+			t.Log("and should fail with error")
+			assert.Equal(t, err, testError)
+			assert.Equal(t, out.String(), "")
```

Comment:
again not super critical to test the empty values on error/failure tests, can just remove this assert